gpflow/base.py

```diff
@@ -26,12 +26,22 @@
     from IPython.lib import pretty
 
 DType = Union[np.dtype, tf.DType]
-VariableData = Union[List[Any], Tuple[Any], np.ndarray, int, float]  # deprecated
+
+if TYPE_CHECKING:
+    # It would be nice to use something more interesting than `Any` here, but it looks like the
+    # infrastructure in the rest of the ecosystem isn't really set up for this yet. Maybe when we
+    # get Python 3.11?
+    AnyNdArray = Union[np.ndarray[Any, Any]]
```

Comment:
I ended up using just writing `"np.ndarray[Any, Any]"` when I needed it (with quotes), which works both at type checking and runtime, but this is probably nicer. Fortunately we have very few ndarrays in trieste, as it's nearly all tensors.

Reply:
Also, to be consistent with `numpy.typing.NDArray`, you may want to name this `AnyNDArray`